### PR TITLE
fix(get_latest_gemini_version): Fix get latest version of gemini

### PR DESCRIPTION
**Pull request by aleksbykov**
Once gemini 1.6.10 was release, the sorting of gemini version work incorrectly
and return wrong latest gemini version

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -242,14 +243,39 @@ def download_file(self, link, dst_dir):
...
     for result_file in results:
-        versions.add(result_file.split('/')[1])
+        gemini_version = [int(num) for num in result_file.split('/')[1].split('.')]
+        if gemini_version in versions:
+            continue
+        versions.append(version)
+
+    sorted_versions = sorted(versions, key=itemgetter(0, 1, 2))
+
+    latest_version = [f"{num}" for num in sorted_versions[-1]]
```
Comment (fruch) [inline sdcm/utils/common.py:276]:
```suggestion
    latest_version = map(str, sorted_versions[-1])
```

File: sdcm/utils/common.py [hunk 2]
```diff
@@ -242,14 +243,39 @@ def download_file(self, link, dst_dir):
...
+    Sort list with version by each element in version and return latest
+    """
     bucket_name = 'downloads.scylladb.com'
 
     results = S3Storage(bucket_name).search_by_path(path='gemini')
-    versions = set()
+    versions = []
     for result_file in results:
-        versions.add(result_file.split('/')[1])
+        gemini_version = [int(num) for num in result_file.split('/')[1].split('.')]
```
Comment (fruch) [inline sdcm/utils/common.py:269]: I wouldn't recommend handling versions like that, or assuming they are always numbers
see
https://stackoverflow.com/a/11887885/459189

and then you can just call sort of the list you get

Comment (aleksbykov) [reply sdcm/utils/common.py:269]: Thanks!

File: sdcm/utils/common.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/utils/common.py]: removed

File: sdcm/utils/common.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/utils/common.py]: fixed

File: sdcm/utils/common.py [hunk 3]
```diff
@@ -242,14 +244,17 @@ def download_file(self, link, dst_dir):
 
 
 def get_latest_gemini_version():
+    """search latest version of gemini tool"""
```
Comment (enaydanov) [inline sdcm/utils/common.py:247]: https://www.python.org/dev/peps/pep-0257/

**tl;dr:** capitalize first letter and add period to the end.

But from my pov docstring is redundant here: `get_latest_gemini_version` is self-explanatory name enough.

File: sdcm/utils/common.py [hunk 4]
```diff
@@ -242,14 +244,17 @@ def download_file(self, link, dst_dir):
...
     bucket_name = 'downloads.scylladb.com'
 
     results = S3Storage(bucket_name).search_by_path(path='gemini')
     versions = set()
     for result_file in results:
-        versions.add(result_file.split('/')[1])
+        versions.add(Version(result_file.split('/')[1]))
+
+    latest_version = sorted(versions)[-1]
```
Comment (enaydanov) [inline sdcm/utils/common.py:255]: Did you thought about `max()`? Looks like it do be doing exactly what happen here.

File: sdcm/utils/common.py [hunk 5]
```diff
@@ -242,14 +244,17 @@ def download_file(self, link, dst_dir):
...
 
 def get_latest_gemini_version():
+    """search latest version of gemini tool"""
     bucket_name = 'downloads.scylladb.com'
 
     results = S3Storage(bucket_name).search_by_path(path='gemini')
     versions = set()
     for result_file in results:
-        versions.add(result_file.split('/')[1])
+        versions.add(Version(result_file.split('/')[1]))
```
Comment (enaydanov) [inline sdcm/utils/common.py:253]: It'll be more pythonic to use list/set comprehension or generator expression instead of the loop here:

```python
versions = {Version(result_file.split("/")[1]) for result_file in results}
```

File: sdcm/utils/common.py [hunk 6]
```diff
@@ -36,12 +36,14 @@
...
 from unittest.mock import Mock
 
+
 from functools import wraps
 from collections import defaultdict
 import concurrent.futures
 from concurrent.futures import ThreadPoolExecutor, TimeoutError as FuturesTimeoutError
 from concurrent.futures.thread import _python_exit
 import hashlib
+from packaging.version import Version
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/utils/common.py:46]: moved

File: sdcm/utils/common.py [hunk 7]
```diff
@@ -36,12 +36,14 @@
 from urllib.parse import urlparse
 from unittest.mock import Mock
 
+
```
[the remark this answers is not in the document]
Comment (aleksbykov) [reply sdcm/utils/common.py:39]: remove

File: sdcm/utils/common.py (hunk 3 above)
Comment (aleksbykov) [reply sdcm/utils/common.py:247]: Removed

File: sdcm/utils/common.py (hunk 5 above)
Comment (aleksbykov) [reply sdcm/utils/common.py]: fixed

File: sdcm/utils/common.py (hunk 4 above)
Comment (aleksbykov) [reply sdcm/utils/common.py]: Thanks, fixed

**Comment by aleksbykov** [comment]
@bentsi @enaydanov Could it be merged and backported soon? gemini jobs should run latest version of gemini

**Comment by bentsi** [comment]
@enaydanov please don't remove backport labels
